spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/JdbcChannelMessageStore.java

```diff
@@ -633,6 +633,10 @@ protected Message<?> doPollForMessage(String groupIdKey) {
 	}
 
 	private boolean doRemoveMessageFromGroup(Object groupId, Message<?> messageToRemove) {
+		if (this.channelMessageStoreQueryProvider.isSingleStatementForPoll()) {
```

Comment:
I don't agree with this expression being present in this method.
It  really does not reflect the purpose of this method.
Please, consider to implement this `if` in the `pollMessageFromGroup()` instead.